lib/loggingService/LogEntry.js

```diff
@@ -247,17 +247,11 @@ LogEntry.prototype.addDetails = function (args) {
 
 
 LogEntry.prototype.addData = function (data) {
-	data = serializeAny(data);
-
-	if (this.data) {
-		for (var key in data) {
-			if (data.hasOwnProperty(key)) {
-				this.data[key] = data[key];
-			}
-		}
-	} else {
-		this.data = data;
+	if (!this.data) {
+		this.data = [];
 	}
+
+	this.data = data.map((entry) => serializeAny(entry)).concat(this.data);
```

Comment:
ah i see, did not know that 勉強になりました ^_^
what about simplifying the line using `...`?
Though this will mean mage will have stricter node version requirements.
@ronkorving thoughts?